internal/dinosql/ktgen.go

```diff
@@ -0,0 +1,986 @@
+
+	case "real", "pg_catalog.float4":
+		return "Float", false
+
+	case "pg_catalog.numeric":
+		return "java.math.BigDecimal", false
+
+	case "bool", "pg_catalog.bool":
+		return "Boolean", false
+
+	case "jsonb":
+		// TODO: support json and byte types
+		return "String", false
+
+	case "bytea", "blob", "pg_catalog.bytea":
+		return "String", false
+
+	case "date":
+		// Date and time mappings from https://jdbc.postgresql.org/documentation/head/java8-date-time.html
+		return "LocalDate", false
+
+	case "pg_catalog.time", "pg_catalog.timetz":
+		return "LocalTime", false
+
+	case "pg_catalog.timestamp":
+		return "LocalDateTime", false
+
+	case "pg_catalog.timestamptz", "timestamptz":
+		// TODO
+		return "OffsetDateTime", false
+
+	case "text", "pg_catalog.varchar", "pg_catalog.bpchar", "string":
+		return "String", false
+
+	case "uuid":
+		// TODO
+		return "uuid.UUID", false
+
+	case "inet":
+		// TODO
+		return "net.IP", false
+
+	case "void":
+		// TODO
+		// A void value always returns NULL. Since there is no built-in NULL
+		// value into the SQL package, we'll use sql.NullBool
+		return "sql.NullBool", false
+
+	case "any":
+		// TODO
+		return "Any", false
+
+	default:
+		for name, schema := range r.Catalog.Schemas {
+			if name == "pg_catalog" {
+				continue
+			}
+			for _, enum := range schema.Enums {
+				if columnType == enum.Name {
+					if name == "public" {
+						return KtDataClassName(enum.Name, settings), true
+					}
+
+					return KtDataClassName(name+"_"+enum.Name, settings), true
+				}
+			}
+		}
+		log.Printf("unknown PostgreSQL type: %s\n", columnType)
+		return "interface{}", false
+	}
+}
+
+func (r Result) ktColumnsToStruct(name string, columns []core.Column, settings CombinedSettings) *KtStruct {
+	gs := KtStruct{
+		Name: name,
+	}
+	seen := map[string]int{}
+	for i, c := range columns {
+		fieldName := KtMemberName(ktColumnName(c, i), settings)
+		if v := seen[c.Name]; v > 0 {
+			fieldName = fmt.Sprintf("%s_%d", fieldName, v+1)
+		}
+		gs.Fields = append(gs.Fields, KtField{
+			Name: fieldName,
+			Type: r.ktType(c, settings),
+		})
+		seen[c.Name]++
+	}
+	return &gs
+}
+
+func ktArgName(name string) string {
+	out := ""
+	for i, p := range strings.Split(name, "_") {
+		if i == 0 {
+			out += strings.ToLower(p)
+		} else {
+			out += strings.Title(p)
+		}
+	}
+	return out
+}
+
+func ktParamName(p Parameter) string {
+	if p.Column.Name != "" {
+		return ktArgName(p.Column.Name)
+	}
+	return fmt.Sprintf("dollar_%d", p.Number)
+}
+
+func ktColumnName(c core.Column, pos int) string {
+	if c.Name != "" {
+		return c.Name
+	}
+	return fmt.Sprintf("column_%d", pos+1)
+}
+
+var jdbcSQLRe = regexp.MustCompile(`\B\$\d+\b`)
+
+// HACK: jdbc doesn't support numbered parameters, so we need to transform them to question marks...
+// But there's no access to the SQL parser here, so we just do a dumb regexp replace instead. This won't work if
+// the literal strings contain matching values, but good enough for a prototype.
```

Comment:
FWIW you shouldn't need a full parser, just a lexer that correctly tokenises strings and comments. Though that in itself might be a challenge across multiple dialects :\